core/src/main/java/com/netflix/conductor/service/WorkflowServiceImpl.java

```diff
@@ -600,4 +603,11 @@ public ExternalStorageLocation getExternalStorageLocation(
                     path);
         }
     }
+
+    private void populateDynamicFlagInSystemMetadata(Map<String, Object> input) {
+        String key = "_system_metadata";
+        Object systemMetadata = input.getOrDefault(key, new HashMap<String, Object>());
+        ((HashMap<String, Object>) systemMetadata).put("dynamic", true);
```

Comment:
Instead of type casting like this, Better we create a map first for metadata, put the dynamic flag, and then put that map into the input. This way it will be better readable and free from type casting issues.